cmd/gencerts.go

```diff
@@ -0,0 +1,147 @@
+package cmd
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"flag"
+	"math/big"
+	"net"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/OpenBazaar/openbazaar-go/repo"
+)
+
+//GenerateCertificates struct
+type GenerateCertificates struct {
+	DataDir  string        `short:"d" long:"datadir" description:"specify the data directory to be used"`
+	Testnet  bool          `short:"t" long:"testnet" description:"config file is for testnet node"`
+	Host     string        `short:"h" long:"host" description:"comma-separated hostnames and IPs to generate a certificate for"`
+	ValidFor time.Duration `long:"duration" description:"duration that certificate is valid for"`
+}
+
+//return PublicKey
+func publicKey(priv interface{}) interface{} {
+	switch k := priv.(type) {
+	case *rsa.PrivateKey:
+		return &k.PublicKey
+	default:
+		return nil
+	}
+}
+
+func pemBlockForKey(priv interface{}) *pem.Block {
+	switch k := priv.(type) {
+	case *rsa.PrivateKey:
+		return &pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(k)}
+	default:
+		return nil
+	}
+}
+
+//Execute gencerts command
+func (x *GenerateCertificates) Execute(args []string) error {
+	// Set repo path
+	repoPath, err := repo.GetRepoPath(x.Testnet)
+	if err != nil {
+		return err
+	}
+	if x.DataDir != "" {
+		repoPath = x.DataDir
+	}
+
+	flag.Parse()
```

Comment:
Is this being called early enough? Seems `x.Testnet` would always be `false` here.